tests/player.c

```diff
@@ -883,6 +891,478 @@ START_TEST (test_play_stream_disable_enable)
 
 END_TEST;
 
+typedef struct ExtSubtitleArgs
+{
+  GstPlayer *player;
+  GMainLoop *loop;
+  gpointer *test_data;
+} ExtSubtitleArgs;
+
+static gboolean
+invalid_external_suburi_test (gpointer user_data)
+{
+  ExtSubtitleArgs *args = (ExtSubtitleArgs *) user_data;
+  gint step = GPOINTER_TO_INT (*(args->test_data));
+
+  fail_unless (gst_player_get_subtitle_uri (args->player) == NULL);
+
+  *(args->test_data) = GINT_TO_POINTER (step + 1);
+  g_main_loop_quit (args->loop);
+
+  return G_SOURCE_REMOVE;
+}
+
+static void
+test_play_error_invalid_external_suburi_cb (GstPlayer * player,
+    TestPlayerStateChange change, TestPlayerState * old_state,
+    TestPlayerState * new_state)
+{
+  gint running = GPOINTER_TO_INT (new_state->test_data);
+
+  if (new_state->state == GST_PLAYER_STATE_PLAYING && !running) {
+    gchar *suburi;
+    ExtSubtitleArgs *args = g_new0 (ExtSubtitleArgs, 1);
+
+    args->player = player;
+    args->loop = new_state->loop;
+    args->test_data = &new_state->test_data;
+    new_state->test_data = GINT_TO_POINTER (running + 1);
+
+    suburi = gst_filename_to_uri (TEST_PATH "/foo.srt", NULL);
+    fail_unless (suburi != NULL);
+
+    /* load invalid suburi */
+    fail_unless (gst_player_set_subtitle_uri (player, suburi) != FALSE);
+    g_free (suburi);
+
+    g_timeout_add_full (G_PRIORITY_DEFAULT, 5000, invalid_external_suburi_test,
```

Comment:
i just looked at gstplayer and it seems like we do not emit warning messages to the app. so to use the warning method we first need to add the emit logic.